packages/pn-commons/src/utility/paymentCaching.utility.ts

```diff
@@ -0,0 +1,133 @@
+import _ from 'lodash';
+
+import { PaymentCache, PaymentDetails, PaymentStatus } from '../models';
+import { paymentCacheSchema } from '../models/PaymentCache';
+
+export const PAYMENT_CACHE_KEY = 'payments';
+
+export const getPaymentCache = (): PaymentCache | null => {
+  const paymentCache = sessionStorage.getItem(PAYMENT_CACHE_KEY);
+  if (!paymentCache) {
+    return null;
+  }
+
+  try {
+    paymentCacheSchema.validateSync(JSON.parse(paymentCache), {
+      stripUnknown: false,
+    });
+  } catch (error) {
+    if (process.env.NODE_ENV === 'development') {
+      console.warn(error);
+    }
+    return null;
+  }
+
+  return paymentCache ? (JSON.parse(paymentCache) as PaymentCache) : null;
```

Comment:
1. due to the if at line 10, this ternary operator is useless.
2. JSON.parse can lead to an exception if json is malformed. use a try catch, similar to above one, to manage this case